testdata/go/v4-alpha/memcached-operator/monitoring/alerts.go

```diff
@@ -0,0 +1,81 @@
+package monitoring
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	ruleName                     = "memcached-operator-rules"
+	alertRuleGroup               = "memcached.rules"
+	deploymentSizeUndesiredAlert = "MemcachedDeploymentSizeUndesired"
+	operatorDownAlert            = "MemcachedOperatorDown"
+	operatorUpTotalRecordingRule = "memcached_operator_up_total"
+)
+
+// NewPrometheusRule creates new PrometheusRule(CR) for the operator to have alerts and recording rules
+func NewPrometheusRule(namespace string) *monitoringv1.PrometheusRule {
+	return &monitoringv1.PrometheusRule{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: monitoringv1.SchemeGroupVersion.String(),
+			Kind:       "PrometheusRule",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ruleName,
+			Namespace: namespace,
+		},
+		Spec: *NewPrometheusRuleSpec(),
+	}
+}
+
+// NewPrometheusRuleSpec creates PrometheusRuleSpec for alerts and recording rules
+func NewPrometheusRuleSpec() *monitoringv1.PrometheusRuleSpec {
+	return &monitoringv1.PrometheusRuleSpec{
+		Groups: []monitoringv1.RuleGroup{{
+			Name: alertRuleGroup,
+			Rules: []monitoringv1.Rule{
+				createDeploymentSizeUndesiredAlertRule(),
+				createOperatorDownAlertRule(),
+				createOperatorUpTotalRecordingRule(),
+			},
+		}},
+	}
+}
+
+// createDeploymentSizeUndesiredAlertRule creates MemcachedDeploymentSizeUndesired alert rule
+func createDeploymentSizeUndesiredAlertRule() monitoringv1.Rule {
+	return monitoringv1.Rule{
+		Alert: deploymentSizeUndesiredAlert,
+		Expr:  intstr.FromString("increase(memcached_deployment_size_undesired_count_total[5m]) >= 3"),
+		Annotations: map[string]string{
+			"description": "Memcached-sample deployment size was not as desired more than 3 times in the last 5 minutes.",
+		},
+		Labels: map[string]string{
+			"severity": "warning",
+		},
```

Comment:
@assafad I think we should add example runbooks to the alerts.

Reply:
Done